packages/framework-core/test/booster.test.ts

```diff
@@ -81,35 +72,145 @@ describe('the `Booster` class', () => {
+          createdAt: random.alphaNumeric(),
+          value: {
+            id: '1',
+            entityID: () => UUID.generate(),
+          } as EventInterface,
+        },
+      ]
+      const providerEventsSearch = fake.returns(searchResult)
+      Booster.configureCurrentEnv((config) => {
+        config.provider = {
+          events: {
+            search: providerEventsSearch,
+          },
+        } as unknown as ProviderLibrary
+        config.events[TestEvent.name] = { class: TestEvent }
+      })
+
+      const eventFilterByType: EventFilterByType = {
+        type: TestEvent.name,
+      }
+      const events = await Booster.events(eventFilterByType)
+
+      for (const event of events) {
+        let eventValue
+        switch (event.type) {
+          case TestEvent.name:
+            eventValue = event.value as TestEvent
+            expect(eventValue.getId()).to.not.throw
+            break
+          default:
+            console.log('Default')
+            break
+        }
+      }
+
+      expect(providerEventsSearch).to.have.been.calledOnce
+    })
   })
-})
 
-describe('the public static function `boosterEventDispatcher`', () => {
-  it('calls `Booster.dispatchEvent` passing the rawEvent', async () => {
-    replace(Booster, 'dispatchEvent', fake())
-    const message = { body: 'Test body' }
+  describe('the public static function `boosterEventDispatcher`', () => {
+    it('calls `Booster.dispatchEvent` passing the rawEvent', async () => {
+      replace(Booster, 'dispatchEvent', fake())
+      const message = { body: 'Test body' }
 
-    await boosterEventDispatcher(message)
+      await boosterEventDispatcher(message)
 
-    expect(Booster.dispatchEvent).to.have.been.calledOnceWith(message)
+      expect(Booster.dispatchEvent).to.have.been.calledOnceWith(message)
+    })
   })
-})
 
-describe('the public static function `boosterServeGraphQL`', () => {
-  it('calls `Booster.serveGraphQL` passing the rawMessage', async () => {
-    replace(Booster, 'serveGraphQL', fake())
-    const message = { body: 'Test body' }
+  describe('the public static function `boosterServeGraphQL`', () => {
+    it('calls `Booster.serveGraphQL` passing the rawMessage', async () => {
+      replace(Booster, 'serveGraphQL', fake())
+      const message = { body: 'Test body' }
 
-    await boosterServeGraphQL(message)
+      await boosterServeGraphQL(message)
 
-    expect(Booster.serveGraphQL).to.have.been.calledOnceWith(message)
+      expect(Booster.serveGraphQL).to.have.been.calledOnceWith(message)
+    })
+  })
+
+  describe('The `entity` method', () => {
+    afterEach(() => {
+      restore()
+    })
+
+    context('given a BoosterConfig', () => {
+      const config = new BoosterConfig('test')
+      config.provider = {} as ProviderLibrary
+
+      it('the `entity` function calls to the `fetchEntitySnapshot` method in the EventStore', async () => {
+        replace(EventStore.prototype, 'fetchEntitySnapshot', fake.returns({ id: '42' }))
```

Comment:
Just to confirm, this `fetchEntitySnapshot` is not the one we wanted to deprecate right? Is it the one from providers?

Reply:
`fetchEntitySnapshot` is a function inside of the class `EventStore`.